google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/it/ITClientSideStatementTest.java

```
@@ -0,0 +1,164 @@
+
+/** Execute ClientSideStatements using the generic connection API. */
+@Category(ParallelIntegrationTest.class)
+@RunWith(JUnit4.class)
+public class ITClientSideStatementTest extends ITAbstractSpannerTest {
+  private static final String SET_AUTOCOMMIT_FALSE = "SET AUTOCOMMIT=FALSE";
+  private static final String SHOW_VARIABLE_AUTOCOMMIT = "SHOW VARIABLE AUTOCOMMIT";
+
+  @Test
+  public void testExecuteUpdateClientSideStatement() {
+    try (Connection connection = createConnection()) {
+      long res = connection.executeUpdate(Statement.of(SET_AUTOCOMMIT_FALSE));
+      assertFalse(connection.isAutocommit());
+      assertEquals(res, 0L);
+      connection.commit();
+      SpannerException e =
+          assertThrows(
+              SpannerException.class,
+              () -> connection.executeUpdate(Statement.of(SHOW_VARIABLE_AUTOCOMMIT)));
+      assertEquals(e.getErrorCode(), ErrorCode.FAILED_PRECONDITION);
+      try (ResultSet rs = connection.executeQuery(Statement.of(SHOW_VARIABLE_AUTOCOMMIT))) {
+        while (rs.next()) {
+          assertFalse(rs.getBoolean(0));
+        }
+      }
+    }
+  }
+
+  @Test
+  public void testExecuteUpdateAsyncClientSideStatement() {
+    try (Connection connection = createConnection()) {
+      ApiFuture<Long> resFuture = connection.executeUpdateAsync(Statement.of(SET_AUTOCOMMIT_FALSE));
+      long res = resFuture.get();
+      assertFalse(connection.isAutocommit());
+      assertEquals(res, 0L);
+      SpannerException e =
+          assertThrows(
+              SpannerException.class,
+              () -> connection.executeUpdateAsync(Statement.of(SHOW_VARIABLE_AUTOCOMMIT)));
+      assertEquals(e.getErrorCode(), ErrorCode.FAILED_PRECONDITION);
+      try (ResultSet rs = connection.executeQuery(Statement.of(SHOW_VARIABLE_AUTOCOMMIT))) {
+        while (rs.next()) {
+          assertFalse(rs.getBoolean(0));
+        }
+      }
+    } catch (ExecutionException | InterruptedException e) {
+      // pass
```

This should cause a test failure and not be silently ignored.